src/aosm/azext_aosm/deploy/artifact.py

```diff
@@ -84,10 +119,52 @@ def _upload_to_storage_account(self, artifact_config: ArtifactConfig) -> None:
                 logger.debug(source_blob.url)
                 self.artifact_client.start_copy_from_url(source_blob.url)
                 logger.info(
-                    "Successfully copied %s from %s to %s", 
-                    source_blob.blob_name, source_blob.account_name, self.artifact_client.account_name
+                    "Successfully copied %s from %s to %s",
+                    source_blob.blob_name,
+                    source_blob.account_name,
+                    self.artifact_client.account_name,
                 )
             else:
                 raise RuntimeError(
                     f"{source_blob.blob_name} does not exist in {source_blob.account_name}."
                 )
+
+    def copy_image(
+        self,
+        cli_ctx,
+        container_registry_client,
+        source_registry_id,
+        source_image,
+        target_registry_resource_group_name,
+        target_registry_name,
+        mode="NoForce",
+    ):
+        target_tags = [source_image]
+
+        source = ImportSource(resource_id=source_registry_id, source_image=source_image)
+
+        import_parameters = ImportImageParameters(
+            source=source,
+            target_tags=target_tags,
+            untagged_target_repositories=[],
+            mode=mode,
+        )
+        try:
+            result_poller = container_registry_client.begin_import_image(
+                resource_group_name=target_registry_resource_group_name,
+                registry_name=target_registry_name,
+                parameters=import_parameters,
+            )
+
+            LongRunningOperation(cli_ctx, "Importing image...")(result_poller)
+
+            logger.info(
+                "Successfully imported %s to %s", source_image, target_registry_name
+            )
+        except Exception as e:
+            logger.error(
+                "Failed to import %s to %s. Check if this image exists in the source registry",
+                source_image,
+                target_registry_name,
+            )
+            logger.debug(e)
```

Comment:
See my other comment about this

Reply:
Do you want the process to carry on after this failure though?

Reply:
I think it might be better catch exceptions, write them out to debug logs and `sys.exit()` at the entrypoint for the CLI rather than here.

Reply:
I think it is ok to carry on. This step is the final step of the `publish` command and you could imagine a case where all the images are ok except the first one and the whole process fails because of it. I don't think that is what we want